src/ConnectedMode/Migration/Wizard/MigrationWizardWindow.xaml.cs

```diff
@@ -46,9 +57,19 @@ private void NavigateToMigrationProgressPage(object sender, RoutedEventArgs e)
 
         private void OnStartMigration(object sender, RoutedEventArgs e)
         {
+            if (migrationInProgress) { return; }
+            migrationInProgress = true;
+
             // Disables all closing / cancel buttons.
             this.migrateButton.Visibility = Visibility.Collapsed;
             this.IsCloseButtonEnabled = false;
+
+            MigrateAsync().Forget();
+        }
+
+        private async Task MigrateAsync()
+        {
+            await connectedModeMigration.MigrateAsync(this, CancellationToken.None);
```

Comment:
I checked out .ContinueWith but using that would make everything a bit ugly. I think it is okay to use await instead here. It is not the exact same functionality but it does not really matter too much here.